distro_test.go

```diff
@@ -72,6 +72,48 @@ func isTestLinuxProcessAlive(d *wsl.Distro) bool {
 	return err == nil
 }
 
+func TestDefaultDistro(t *testing.T) {
+	want := newTestDistro(t, emptyRootFs)
+
+	// Not linting G204: Subprocess launched with a potential tainted input or cmd arguments.
+	// No threat of code injection, want.Name is generated by func UniqueDistroName.
+	out, err := exec.Command("wsl.exe", "--set-default", want.Name).CombinedOutput() //nolint:gosec
+	require.NoErrorf(t, err, "setup: failed to set test distro as default: %s", out)
+
+	got, err := wsl.DefaultDistro()
+	require.NoError(t, err, "unexpected error getting default distro %q", want.Name)
+	require.Equal(t, want, got, "Unexpected mismatch in default distro")
+}
+
+func TestDistroSetAsDefault(t *testing.T) {
+	realDistro := newTestDistro(t, emptyRootFs)
+	fakeDistro := wsl.Distro{Name: "This distro sure does not exist"}
+
+	testCases := map[string]struct {
+		distro  wsl.Distro
+		wantErr bool
+	}{
+		"set an existing distro as default":          {distro: realDistro},
+		"fail to set non-existent distro as default": {distro: fakeDistro, wantErr: true},
```

Comment:
I see you like to say fail, but it does return an error, this is really a nitpick of mine feel free to ignore :)

Reply:
We understand failure as an unreported error, then?

Reply:
To me, yes, but it can be a french thingy :p

Reply:
Haha well it makes sense so let's stick to this nomenclature.